crates/sp-domains-fraud-proof/src/fraud_proof.rs

```diff
@@ -130,11 +187,18 @@ impl ExecutionPhase {
             }
             ExecutionPhase::ApplyExtrinsic {
                 extrinsic_proof: proof_of_inclusion,
-                mismatch_index,
+                mismatch,
             } => {
+                let mismatch_index = match mismatch {
+                    ApplyExtrinsicMismatch::StateRoot(mismatch_index) => *mismatch_index,
+                    ApplyExtrinsicMismatch::Shorter => (bad_receipt.execution_trace.len() - 1)
+                        .try_into()
+                        .expect("trace index should always fit into u32; qed"),
```

Comment:
Size of usize type dependent upon the architecture on which node is running. on 64-bit architecture usize can be of 8 bytes and on 32-bit it can be 4 bytes.

So, it is idiomatic to have try_into and except or fail in case value of usize variable does not fit into u32.

Reply:
Emmm I don't think that will be a concern unless `execution_trace.len() > u32::MAX`, `xx as u32` are all over the codebase actually.

Reply:
I understand that this is not the concern, but I would prefer to write cautious code in all situations, unless there are any performance benefit or less amount of code, either of those is not the case here.

Reply:
`bad_receipt.execution_trace.len() as u32 - 1` is certainly cleaner and less loc than
```rust
(bad_receipt.execution_trace.len() - 1)
                        .try_into()
                        .expect("trace index should always fit into u32; qed"),
```
I don't think it is safer either, if `execution_trace.len() > u32::MAX` happen it will occupy 256GB memory, and `xx as u32` is not only all over our codebase but also the upstream.